Code/Framework/AzCore/AzCore/DOM/DomUtils.h

```diff
@@ -307,7 +318,20 @@ namespace AZ::Dom::Utils
                 if (value.IsOpaqueValue())
                 {
                     const AZStd::any& opaqueValue = value.GetOpaqueValue();
-                    if (!opaqueValue.is<WrapperType>())
+                    // If the wrapper type is a reference wrapper, then the original type needs to be extracted
+                    // from the opaque object and then constructed into a reference_wrapper.
+                    // This is because the implementation of std::reference_wrapper might store a pointer to T or a reference to T
+                    // and that cannot be relied on
+                    if constexpr (AZStd::is_reference_wrapper<WrapperType>::value)
+                    {
+                        if (auto instanceValue = AZStd::any_cast<AZStd::remove_reference_t<T>>(&opaqueValue);
+                            instanceValue != nullptr)
+                        {
+                            // Construct a reference_wrapper using a deferenced value to the opaque type
+                            return WrapperType(const_cast<T>(*instanceValue));
+                        }
```

Comment:
The next change needed to allow ValueFromType to properly convert a `reference_wrapper<T>` to a `reference_wrapper<T>` for the identity conversion is add another if block here to check if the opaque value is actually storing a WrapperType.
```suggestion
                        if (auto instanceValue = AZStd::any_cast<AZStd::remove_reference_t<T>>(&opaqueValue);
                            instanceValue != nullptr)
                        {
                            // Construct a reference_wrapper using a deferenced value to the opaque type
                            return WrapperType(const_cast<T>(*instanceValue));
                        }
                        // Check if The Dom Value is actually storing an reference_wrapper<T> and return that if possible
                        else if (auto referenceWrapperValue = AZStd::any_cast<WrapperType>(&opaqueValue);
                            referenceWrapperValue != nullptr)
                        {
                            // return the reference_wrapper<T> directly
                            return referenceWrapperValue;
                        }
```